### ISSUE TITLE
Compiler should be able to infer type of null in super-call

### ISSUE THREAD

**Issue by Frotty** [peq/WurstScript#389]
```
public abstract class OnExpireEvent
    abstract function onExpire(vec2 pos, player owner)

public class PhysicsProjectile extends Projectile
    use PhysicsModule
    OnExpireEvent oe
    construct( vec3 pos, real radius, player owner, angle xyAngle, string fxpath, OnExpireEvent onEvent )

...

class Rocket extends PhysicsProjectile

    construct(vec3 startPos, vec3 targetPos)
        super(startPos, 16, DUMMY_PLAYER, angle(0), "Abilities\\Weapons\\DragonHawkMissile\\DragonHawkMissile.mdl", null )
```

Line 14: Cannot use 'null' constant here because the compiler cannot infer which kind of null it is.

**Comment by Cokemonkey11** [comment]
:+1:

**Comment by peq** [comment]
Should be fixed with b4e21e0c73660d610853a8155d31a4c3fb5d061f
